addons/website_slides/static/src/js/public/components/slide_quiz_finish_dialog/slide_xp_progress_bar.js

```diff
@@ -0,0 +1,102 @@
+/** @odoo-module **/
+
+import { Component, onMounted, useState } from "@odoo/owl";
+import { browser } from "@web/core/browser/browser";
+
+export class SlideXPProgressBar extends Component {
+    static props = {
+        previousRank: Object,
+        newRank: Object,
+        levelUp: Boolean,
+    };
+    static template = "website_slides.SlideXPProgressBar";
+
+    setup() {
+        super.setup();
+        this.state = useState({
+            hidePreviousRankBounds: true,
+            lower_bound: this.props.previousRank.lower_bound,
+            progressBarWidth: this.props.previousRank.progress,
+            progressBarNoTransition: false,
+            tooltipValue: this.props.previousRank.karma,
+            tooltipLeft: this.props.previousRank.progress,
+            upper_bound: this.props.previousRank.upper_bound,
+        });
+        onMounted(() => {
+            this._animateProgressBar();
+        });
+    }
+
+    //--------------------------------
+    // Private
+    //--------------------------------
+
+    /**
+     * Handles the animation of the karma gain in the following steps:
+     * 1. Animate the tooltip text to increment smoothly from the old
+     *    karma value to the new karma value and updates it to make it
+     *    move as the progress bar moves.
+     * 2a. The user doesn't level up
+     *    I.   When the user doesn't level up the progress bar simply goes
+     *         from the old karma value to the new karma value.
+     * 2b. The user levels up
+     *    I.   The first step makes the progress bar go from the old karma
+     *         value to 100%.
+     *    II.  The second step makes the progress bar go from 100% to 0%.
+     *    III. The third and final step makes the progress bar go from 0%
+     *         to the new karma value. It also changes the lower and upper
+     *         bound to match the new rank.
+     * @private
+     */
+    _animateProgressBar() {
```

Comment:
Ah yes, I remember this...

Have to say having a full Component + a template, and some "complex" logic with recursive methods just for an animation feels a bit overkill ^^

I'd like to simplify it.
Goal would be to keep the same behavior (or something close) by using more simple tricks.

One thing that comes to mind is to try to do the same thing using css:
- You could change the "width" of the bar and have a simple css transition rule
- For the counter, there seems to exist something as well, looking on the internet: https://css-tricks.com/animating-number-counters/#the-new-school-css-solution (did not check browser compatibility though) 

Can I ask you to give it a look? Would be great to simplify this while we're re-working it.